design/marketplace.md

```diff
@@ -162,16 +162,99 @@ The client is able to retrieve any funds that are left in the Request.
     ---------
 
 
+Slot reservations
+-----------------
+
+Competition between hosts to fill slots has some advantages, such as providing
+an incentive for hosts to become proficient in downloading content and
+generating proofs. It also has some drawbacks, for instance it can lead to
+network inefficiencies because multiple hosts do the work of downloading and
+proving, while only one host is rewarded for it. These inefficiencies lead to
+higher costs for hosts, which leads to an overall increase in the price of
+storage on the network. It can also lead to clients inadvertently inviting too
+much network traffic to themselves. Should they for instance post a very
+lucrative storage request, then this invites a lot of hosts to start downloading
+the content from the client simultaneously, not unlike a DDOS attack.
+
+Slot reservations are a means to avoid these inefficiencies. Before downloading
+the content associated with a slot, a host can deposit collateral to reserve the
+slot. When it succeeds in reserving the slot, then no other hosts can fill the
+slot. After the host downloads the content and calculates a proof, it moves the
+slot from its reserved state into the filled state. Then it begins to
+periodically provide storage proofs and accrue payments for the slot.
+
+        collateral               proof
+            |                      |
+            v                      v
+            ------------------------------------------------------------------
+     slot:  |/ / / / / / / / / / / |//////////////////////////////////////////
+            ------------------------------------------------------------------
+            |                      |
+            v                      v
+          slot                    slot
+         reserved                filled
+
+
+            ---------------- time ---------------->
+
+
+Slot reservations could however provide a relatively cheap way for malicious
```

Comment:
I think overall, these approaches have severe UX drawbacks and they expose both the client and the providers to serious loss, both in monetary and resources. I did an analysis around the initial [contract interaction](https://github.com/codex-storage/codex-research/discussions/166). I believe approach 2 from my writeup is the most flexible and has the best UX characteristics, we should seriously re-evaluate this approach in the same context.

Reply:
Thanks for the detailed write-up. I think you're right. Allowing a host to reserve a slot without collateral makes the whole design simpler. Squatting of slots, which is the main downside of this approach, is discouraged by both the gas fees, and the expanding window mechanism. I think those should be enough of a deterrent. I have updated the document to reflect this design.

Reply:
Adam raised some good points about the strength of the deterrents to prevent squatting. If we could disincentivise client withholding, then perhaps approach 3 would be the best. I outlined a suggestion [in the discussion](https://github.com/codex-storage/codex-research/discussions/166#discussioncomment-6106603).